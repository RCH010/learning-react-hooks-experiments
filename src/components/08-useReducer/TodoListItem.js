import React from 'react'
import PropTypes from 'prop-types'

export const TodoListItem = ({todo, index, handleDelete, handleToggle}) => {
    return (
        <li
            key={index}
            className='list-group-item'
        >
            <p
                className={`${todo.done && 'complete'}`}
                onClick={ () => handleToggle(todo.id)}
            >
                {index+1}. {todo.desc}
            </p>
            <button 
                className='btn btn-danger'
                onClick={() => handleDelete(todo.id)}
            >
                Borrar
            </button>
        </li>
    )
}

TodoListItem.propTypes = {
    todo: PropTypes.object.isRequired,
    index: PropTypes.number.isRequired,
    handleDelete: PropTypes.func.isRequired,
    handleToggle: PropTypes.func.isRequired,
}

export default TodoListItem
