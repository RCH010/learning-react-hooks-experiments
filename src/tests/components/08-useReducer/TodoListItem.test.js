import React from 'react';
import { shallow } from "enzyme"
import { demoTodos } from "../../fixitures/demoTodos"
import TodoListItem from "../../../components/08-useReducer/TodoListItem"

describe('Tests in <TodoListItem />', () => {
    const handelDelete = jest.fn();
    const handleToggle = jest.fn();

    const wrapper = shallow (
        <TodoListItem 
            todo = {demoTodos[0]}
            index = {1}
            handleDelete = { handelDelete }
            handleToggle = { handleToggle }
        />)

    test('should match with snapshot, all default', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('should call errase function', () => {
        wrapper.find('button').simulate('click');
        expect(handelDelete).toHaveBeenCalledWith(demoTodos[0].id);
    })

    test('should call handleToggle function', () => {
        wrapper.find('p').simulate('click');
        expect(handleToggle).toHaveBeenCalledWith(demoTodos[0].id);
    })

    test('should display todo correctly', () => {
        const text = wrapper.find('p').text();
        expect(text).toBe(`2. ${demoTodos[0].desc}`);
    })
    
    test('should have class complete, when todo.done is true', () => {
        const todo = demoTodos[0];
        todo.done = true;

        const wrapper = shallow (
            <TodoListItem 
                todo = {todo}
                index = {1}
                handleDelete = { handelDelete }
                handleToggle = { handleToggle }
            />)
        expect(wrapper.find('p').hasClass('complete')).toBe(true);
    })
})
