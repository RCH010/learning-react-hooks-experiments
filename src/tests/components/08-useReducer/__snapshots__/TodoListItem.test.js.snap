// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <TodoListItem /> should match with snapshot, all default 1`] = `
<li
  className="list-group-item"
  key="1"
>
  <p
    className="false"
    onClick={[Function]}
  >
    2
    . 
    Aprender React
  </p>
  <button
    className="btn btn-danger"
    onClick={[Function]}
  >
    Borrar
  </button>
</li>
`;
